extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -111,6 +114,20 @@ function VideoWrapperWithRef(
   // <source>s change.
   const readyDeferred = useMemo(() => new Deferred(), []);
 
+  const [readyState, setReadyState_] = useState(ReadyState.LOADING);
+
+  const onReadyStateRef = useValueRef(onReadyState);
+  const setReadyState = useCallback(
```

Comment:
It's not very idiomatic, but it happens. The only reason it's done here is to avoid re-firing effects that depend on `setReadyState` because those would indicate "logical" state changes and changing the event handler doesn't constitute a state change. Especially because inlining events is super common in React. E.g. `<Comp onReadyState={() => ...}>`.

Reply:
Which effect would be refired when `onReadyState` is modified? All I see is that the `setReadyState` callback would be redefined.

Reply:
[This one](https://github.com/ampproject/amphtml/pull/32974/files#diff-0f225f57ef9d0e91c443646021b7b37948b85772c99a6b9a74b90223d1c2a650R159)

```
  useLayoutEffect(() => {
    const player = playerRef.current;
    const readyState = player && player.readyState;
    if (readyState != null) {
      setReadyState(readyState > 0 ? ReadyState.COMPLETE : ReadyState.LOADING);
    }
  }, [setReadyState]);
```

We can get into a weird loop here. Say the component is called with inlined `onReadyState` callback like this: `<Comp onReadyState={() => ...}>`. And let's take a case where `onReadyState` can cause re-render. What we get then here is this sequence:

1. The parent component renders `<Comp onRenderState={() => }>`.
2. The `setRenderState` callback is rebuilt.
3. `useLayoutEffect(() => {}, [setRenderState])` is triggered.
4. The parent's `onRenderState` callback is called. It causes the re-render.
5. The parent re-renders `<Comp>` with new `onReadyState`.
6. Repeat.